<?xml version="1.0" ?>
<launch>
  <arg name="config"		default="aist"/>
  <arg name="scene"		default="$(arg config)_base_scene"/>
  <arg name="sim"		default="false"/>
  <arg name="rviz_config"
       default = "$(find aist_moveit_config)/launch/moveit.rviz"/>
  <arg name="config_dir"        default="$(find o2as_fastening_tools)/config"/>
  <arg name="suction_control"   default="suction_control"/>
  <arg name="operation_mode"    default="operation_mode"/>

  <param name="use_real_robot"	value="$(eval not arg('sim'))"/>

  <!-- Launch real robots if "sim" == "false" -->
  <include file="$(find o2as_examples)/launch/connect_real_robots.launch"
	   if="$(eval arg('config') == 'o2as' and not arg('sim'))"/>
  <include file="$(find aist_examples)/launch/connect_$(arg config)_robots.launch"
	   if="$(eval not arg('config') == 'o2as' and not arg('sim'))"/>

  <!-- Launch planner -->
  <include file="$(find o2as_moveit_config)/launch/o2as_moveit_planning_execution.launch"
  	   if="$(eval arg('config') == 'o2as')">
    <arg name="sim" value="$(arg sim)"/>
  </include>
  <include file="$(find aist_moveit_config)/launch/aist_moveit_planning_execution.launch"
	   unless="$(eval arg('config') == 'o2as')">
    <arg name="config"		value="$(arg config)"/>
    <arg name="scene"		value="$(arg scene)"/>
    <arg name="sim"		value="$(arg sim)"/>
    <arg name="rviz_config"	value="$(arg rviz_config)"/>
  </include>

  <!-- Launch skill server -->
  <include file="$(find o2as_skills)/launch/o2as_skill_server.launch"
	   if="$(eval arg('config') == 'o2as')"/>

  <!-- Launch fastening tools -->
  <node name="suction_controller" pkg="o2as_fastening_tools"
	type="suction_controller.py" output="screen"
	if="$(eval not arg('config') == 'o2as' and not arg('sim'))">
    <param name="config_dir"      type="string" value="$(arg config_dir)"/>
    <param name="suction_control" type="string" value="$(arg suction_control)"/>
    <param name="operation_mode"  type="string" value="$(arg operation_mode)"/>
  </node>

</launch>
